src/test/java/codeu/controller/ChatServletTest.java

```diff
@@ -196,4 +196,58 @@ public void testDoPost_CleansHtmlContent() throws IOException, ServletException
 
     Mockito.verify(mockResponse).sendRedirect("/chat/test_conversation");
   }
+  
+  @Test
+  public void testDoPost_ModifyMessageBoldItalics() throws IOException, ServletException{
+    Mockito.when(mockRequest.getRequestURI()).thenReturn("/chat/test_conversation");
+    Mockito.when(mockSession.getAttribute("user")).thenReturn("test_username");
+
+    User fakeUser = new User(UUID.randomUUID(), "test_username", "password", Instant.now());
+    Mockito.when(mockUserStore.getUser("test_username")).thenReturn(fakeUser);
+
+    Conversation fakeConversation =
+        new Conversation(UUID.randomUUID(), UUID.randomUUID(), "test_conversation", Instant.now());
+    Mockito.when(mockConversationStore.getConversationWithTitle("test_conversation"))
+        .thenReturn(fakeConversation);
+
+    Mockito.when(mockRequest.getParameter("message"))
+        .thenReturn("This sentence ***tests*** that **bold**and*italics* work \\**correctly**.");
+
+    chatServlet.doPost(mockRequest, mockResponse);
+
+    ArgumentCaptor<Message> messageArgumentCaptor = ArgumentCaptor.forClass(Message.class);
+    Mockito.verify(mockMessageStore).addMessage(messageArgumentCaptor.capture());
+    Assert.assertEquals(
+        "This sentence <b><i>tests</i></b> that <b>bold</b>and<i>italics</i> work *<i>correctly</i>*."
+        , messageArgumentCaptor.getValue().getContent());
+
+    Mockito.verify(mockResponse).sendRedirect("/chat/test_conversation");
+  }
+  
+  @Test
+  public void testDoPost_ModifyMessageColorSize() throws IOException, ServletException{
+    Mockito.when(mockRequest.getRequestURI()).thenReturn("/chat/test_conversation");
+    Mockito.when(mockSession.getAttribute("user")).thenReturn("test_username");
+
+    User fakeUser = new User(UUID.randomUUID(), "test_username", "password", Instant.now());
+    Mockito.when(mockUserStore.getUser("test_username")).thenReturn(fakeUser);
+
+    Conversation fakeConversation =
+        new Conversation(UUID.randomUUID(), UUID.randomUUID(), "test_conversation", Instant.now());
+    Mockito.when(mockConversationStore.getConversationWithTitle("test_conversation"))
+        .thenReturn(fakeConversation);
+
+    Mockito.when(mockRequest.getParameter("message"))
+        .thenReturn("[size=4]glue[/size] [color=green]\\[color=blue]this is blue\\[/color][/color]");
+
+    chatServlet.doPost(mockRequest, mockResponse);
+
+    ArgumentCaptor<Message> messageArgumentCaptor = ArgumentCaptor.forClass(Message.class);
+    Mockito.verify(mockMessageStore).addMessage(messageArgumentCaptor.capture());
+    Assert.assertEquals(
+        "<font size=\"4\">glue</font> <font color=\"green\">[color=blue]this is blue[/color]</font>"
+        , messageArgumentCaptor.getValue().getContent());
+
```

Comment:
good job capturing all the expected behaviour for color/size with one test string!